MediaBrowser.Model/Dto/MediaSourceInfo.cs

```diff
@@ -22,33 +22,34 @@ public MediaSourceInfo()
             SupportsDirectStream = true;
             SupportsDirectPlay = true;
             SupportsProbing = true;
+            TranscodeReasons = Array.Empty<TranscodeReason>();
         }
 
         public MediaProtocol Protocol { get; set; }
 
-        public string Id { get; set; }
+        public string? Id { get; set; }
```

Comment:
This is a big no-no combined with the use of the `!` operator. Either it can be null or it can't. Generally we shouldn't be using `!`.